modules/apps/dynamic-data-mapping/dynamic-data-mapping-test/src/testIntegration/java/com/liferay/dynamic/data/mapping/info/item/provider/test/DDMFormValuesInfoFieldValuesProviderTest.java

```diff
@@ -105,6 +122,44 @@ public void testGetInfoFieldValuesCheckboxMultipleDDMFormFieldType()
 				).build()));
 	}
 
+	@Test
+	public void testGetInfoFieldValuesNestedStructure() throws Exception {
+		DDMStructure ddmStructure = _addDDMStructure(
+			_group, "test-structured-content-nested-structure.json");
+
+		long classNameId = ClassNameLocalServiceUtil.getClassNameId(
+			JournalArticle.class);
+
+		ServiceContext serviceContext =
+			ServiceContextTestUtil.getServiceContext(
+				_group.getGroupId(), TestPropsValues.getUserId());
+
+		JournalFolder journalFolder = _addJournalFolder(
+			ddmStructure.getStructureId(), serviceContext);
+
+		DDMTemplate ddmTemplate = DDMTemplateTestUtil.addTemplate(
+			_group.getGroupId(), ddmStructure.getStructureId(), classNameId);
+
+		String content = _read("ddm-structure-content-nested.xsd");
+
+		Map<Locale, String> nameMap = HashMapBuilder.put(
+			LocaleUtil.US, RandomTestUtil.randomString()
+		).build();
+
+		JournalArticle journalArticle = _journalArticleLocalService.addArticle(
+			null, TestPropsValues.getUserId(), _group.getGroupId(),
+			journalFolder.getFolderId(), nameMap, nameMap, content,
+			ddmStructure.getStructureId(), ddmTemplate.getTemplateKey(),
+			serviceContext);
+
+		List<InfoFieldValue<InfoLocalizedValue<Object>>> infoFieldValues =
+			_ddmFormValuesInfoFieldValuesProvider.getInfoFieldValues(
+				journalArticle, journalArticle.getDDMFormValues());
+
+		Assert.assertEquals(
+			infoFieldValues.toString(), 6, infoFieldValues.size());
```

Comment:
I saw that the other methods in this class are testing the items values and here you are just asserting the size. Could you assert the values? Why did you decide just to assert the size?

Reply:
Would be something like this?

for(DDMFormField ddmFormField : ddmStructure.getDDMFormFields(true)){
			if(!ddmFormField.getName().contains("FieldSet")){
				_assertGetInfoFieldValues(
					ddmFormField.getName(), journalArticle,
					Assert::assertNull);
			}
		}
		
		The _assertGetInfoFieldValues doesn't support nested fields for now, so I thought in order to match the expected results of the ticket just to know if all the fields were being returned, i.e. the quantity

Reply:
Yeah, it can be something like this.
If you use ddmStructure.getDDMFormFields(**true**), I believe the fieldset type will not be included in that list. This happens because a transient field is a field that does not have value (for example, fieldset, separator).

Reply:
I believe you're saying false, is that right? Really good to know this!! Thank you for the amazing tip.